variables.tf

```diff
@@ -93,81 +35,23 @@ variable "location" {
 
 variable "lock" {
   type = object({
+    kind = string
     name = optional(string, null)
-    kind = optional(string, "None")
   })
-  default     = {}
-  description = "The lock level to apply. Default is `None`. Possible values are `None`, `CanNotDelete`, and `ReadOnly`."
-  nullable    = false
+  default     = null
+  description = <<DESCRIPTION
+  Controls the Resource Lock configuration for this resource. The following properties can be specified:
```

Comment:
Maybe would be good to mention the lock object itself is `(Optional)`

Reply:
This was copied from the required interfaces: https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#resource-locks

Let me know if this is still required :)